docs/docs/guides/integrate/identity-providers/okta_saml.mdx

```diff
@@ -0,0 +1,103 @@
+
+As an alternative you can add the SAML identity provider trhough the API, either on the default settings or on a specific organization:
+- [Add Default SAML Identity Provider](/docs/apis/resources/admin/admin-service-add-saml-provider)
+- [Add SAML Identity Provider on Organization](/docs/apis/resources/mgmt/management-service-add-saml-provider)
+
+![OKTA Provider Empty](/img/guides/zitadel_okta_saml_provider_empty.png)
+
+## OKTA Configuration
+
+### Register a new client
+
+1. Login to your OKTA Account and go to the applications list: <OKTA-DOMAIN/admin/apps/active>
+2. Click on "Create App Integration" and choose "SAML 2.0"
+3. Give the application a name
+4. Fill the configuration as following (Replace `your-domain` and `saml-idp-id` with your data):
+ - Single sign-on URL {your-domain}/ui/login/login/externalidp/saml/acs
+ - Audience URI (SP Entity ID): {your-domain}/idps/{saml-idp-id}/saml/metadata
+ - Example redirect url for the domain `https://acme-gzoe4x.zitadel.cloud` would look like this:  `https://acme-gzoe4x.zitadel.cloud/idps/257372385775925924/saml/metadata`
+5. Save the configuration
+6. Copy the metadata URL from the details
+
+![Add new SAML Application in OKTA](/img/guides/okta_add_saml_app.png)
+
+### Assign Users to Application
+
+To allow users to authenticate with that app go to the "Assign" Tab.
+1. Click the Assign Button
+2. Choose Assign To People
+3. Select the users you like to be able to authenticate
+
+![Add new SAML Application in OKTA](/img/guides/okta_assign_user_to_app.png)
+
+## Finish ZITADEL Configuration
+
+You are now finished with the configuration in OKTA and you can switch back to your identity provider configuration in ZITADEL.
+
+### Add Metadata Xml
+
+Add the metadata URL you have saved before from OKTA to the Metadata URL.
+As soon as you have saved the provider, and you have a look at the detail you should now see the Metadata Xml field filled.
+
+If you prefere changing the configuration through the API you can update the SAML provider on the default settings or a specific orgnaization:
+- [Update Default SAML Identity Provider](/docs/apis/resources/admin/admin-service-update-saml-provider)
+- [Update SAML Identity Provider on Organization](/docs/apis/resources/mgmt/management-service-update-saml-provider)
+
+![OKTA Provider Empty](/img/guides/zitadel_okta_saml_provider_filled.png)
+
+You can also fill the optional fields if needed:
+
+<GeneralConfigDescription provider_account="OKTA account" />
+
+### Activate IdP
+
+<Activate/>
+
+![Activate the OKTA Provider](/img/guides/zitadel_activate_okta_saml.png)
+
+## Test the setup
+
+<TestSetup loginscreen="your OKTA login"/>
+
+![OKTA Button](/img/guides/zitadel_login_okta.png)
+
+![OKTA Login](/img/guides/okta_login.png)
+
+:::note
+At the moment it is not possible to autofill the user data in ZITADEL with the data from OKTA.
+This means the user has to fill the information manually.
```

Comment:
Does okta not provide any information or is an example action missing?

Reply:
it is not possible with the actions we currently have as far as i know

Reply:
I just tested it with entra id. it should be possible: https://github.com/zitadel/actions/pull/31/files